File: C/Problem0012.c
/**
 * Problem0012 - Highly divisible triangular number
 * 
 * https://github.com/agileshaw/Project-Euler
 **/

#include <stdio.h>

int numOfFactors(unsigned long long num)
{
    int count = 0; 
    unsigned long long i;

    for (i = 1; i <= num; i++) {
        if (num % i == 0)
            count++;
    }

    return count;
}

int main ()
{
    int count = 0;
    unsigned long long natural = 1, result = 0;

    while (count < 500) {
        
        result += natural;
        natural++;
        printf("%llu\n", result);
        count = numOfFactors(result);
        }

    printf("The highly divisible triangular number is: %llu\n", result);

    return 0;
}

File: Perl/Problem0012.pl
# Problem0012 - Highly divisible triangular number
#
# https://github.com/agileshaw/Project-Euler

use strict;
use warnings;

my $count = 0;
my $natural = 1;
my $result = 0;
sub numOfFactors;

while ($count < 500) {
    $result += $natural;
    $natural++;
    $count = numOfFactors($result);
}

print "The highly divisible triangular number is: $result\n";

sub numOfFactors
{
    my $num = shift;
    my $root = int(sqrt($num));
    my $count = 0;

    for my $i (1..$root) {
        if ($num % $i == 0) {
            $count += 2;
        }
    }

    if ($root * $root == $num) {
        $count--;
    }

    return $count;
}